### PR TITLE
Allow export clientInfo, userAgent and spark queue in Presto-on-Spark

### PR DESCRIPTION
**Pull request by wenleix**
userAgent can be used to store Spark application id.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java [hunk 1]
```diff
@@ -215,7 +217,9 @@ public PrestoSparkQueryExecutionFactory(
     public IPrestoSparkQueryExecution create(
             SparkContext sparkContext,
             PrestoSparkSession prestoSparkSession,
+            Optional<String> userAgent,
```
Comment (arhimondr) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:220]: Let's pass it through the `PrestoSparkSessionContext` / `PrestoSparkSession`. The `SessionContext` interface has the

```
    @Nullable
    String getUserAgent();
```

declared. We just need to implement it in the `PrestoSparkSessionContext` and in the `PrestoSparkSession`

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java [hunk 2]
```diff
@@ -426,6 +435,8 @@ private static QueryInfo createQueryInfo(
                 succinctBytes(peakTaskUserMemoryInBytes),
                 succinctBytes(peakTaskTotalMemoryInBytes));
 
+        Optional<ResourceGroupId> resourceGroupId = sparkQueueName.map(name -> new ResourceGroupId(Arrays.asList(name.split("\\."))));
```
Comment (arhimondr) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:438]: Do you think it is necessary to split it on `"\\."`? Shouldn't we just store it as is in a single fragment?

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java (hunk 1 above)
Comment (wenleix) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:220]: @arhimondr : it's the other way around, `PrestoSparkSessionContext` is created in `PrestoSparkQueryExecutionFactory#create`. Thus the user agent need to passed in here (so `PrestoSparkSessionContext` will have the user agent we want

File: presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java (hunk 2 above)
Comment (wenleix) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java:436]: @arhimondr : Many years ago there is a discussion with @raghavsethi  and we agreed "." shouldn't be a valid character in resource group name :) ... 

anyhow, assume it doesn't cause trouble, I am happy to just store it in a single fragment.

**Comment by arhimondr** [comment]
LGTM

**Pull request merged**